File: android/app/src/main/java/gg/strims/ppspp/rpc/BootstrapClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class BootstrapClient(filepath: String) : RPCClient(filepath) {

    suspend fun createClient(
        arg: CreateBootstrapClientRequest = CreateBootstrapClientRequest()
    ): CreateBootstrapClientResponse =
        this.callUnary("Bootstrap/CreateClient", arg)

    suspend fun updateClient(
        arg: UpdateBootstrapClientRequest = UpdateBootstrapClientRequest()
    ): UpdateBootstrapClientResponse =
        this.callUnary("Bootstrap/UpdateClient", arg)

    suspend fun deleteClient(
        arg: DeleteBootstrapClientRequest = DeleteBootstrapClientRequest()
    ): DeleteBootstrapClientResponse =
        this.callUnary("Bootstrap/DeleteClient", arg)

    suspend fun getClient(
        arg: GetBootstrapClientRequest = GetBootstrapClientRequest()
    ): GetBootstrapClientResponse =
        this.callUnary("Bootstrap/GetClient", arg)

    suspend fun listClients(
        arg: ListBootstrapClientsRequest = ListBootstrapClientsRequest()
    ): ListBootstrapClientsResponse =
        this.callUnary("Bootstrap/ListClients", arg)

    suspend fun listPeers(
        arg: ListBootstrapPeersRequest = ListBootstrapPeersRequest()
    ): ListBootstrapPeersResponse =
        this.callUnary("Bootstrap/ListPeers", arg)

    suspend fun publishNetworkToPeer(
        arg: PublishNetworkToBootstrapPeerRequest = PublishNetworkToBootstrapPeerRequest()
    ): PublishNetworkToBootstrapPeerResponse =
        this.callUnary("Bootstrap/PublishNetworkToPeer", arg)

}

File: src/lib/api/bootstrapClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class Bootstrap {
  constructor(private readonly host: RPCHost) {}

  public createClient(
    arg: pb.ICreateBootstrapClientRequest = new pb.CreateBootstrapClientRequest()
  ): Promise<pb.CreateBootstrapClientResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/CreateClient", new pb.CreateBootstrapClientRequest(arg))
    );
  }
  public updateClient(
    arg: pb.IUpdateBootstrapClientRequest = new pb.UpdateBootstrapClientRequest()
  ): Promise<pb.UpdateBootstrapClientResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/UpdateClient", new pb.UpdateBootstrapClientRequest(arg))
    );
  }
  public deleteClient(
    arg: pb.IDeleteBootstrapClientRequest = new pb.DeleteBootstrapClientRequest()
  ): Promise<pb.DeleteBootstrapClientResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/DeleteClient", new pb.DeleteBootstrapClientRequest(arg))
    );
  }
  public getClient(
    arg: pb.IGetBootstrapClientRequest = new pb.GetBootstrapClientRequest()
  ): Promise<pb.GetBootstrapClientResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/GetClient", new pb.GetBootstrapClientRequest(arg))
    );
  }
  public listClients(
    arg: pb.IListBootstrapClientsRequest = new pb.ListBootstrapClientsRequest()
  ): Promise<pb.ListBootstrapClientsResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/ListClients", new pb.ListBootstrapClientsRequest(arg))
    );
  }
  public listPeers(
    arg: pb.IListBootstrapPeersRequest = new pb.ListBootstrapPeersRequest()
  ): Promise<pb.ListBootstrapPeersResponse> {
    return this.host.expectOne(
      this.host.call("Bootstrap/ListPeers", new pb.ListBootstrapPeersRequest(arg))
    );
  }
  public publishNetworkToPeer(
    arg: pb.IPublishNetworkToBootstrapPeerRequest = new pb.PublishNetworkToBootstrapPeerRequest()
  ): Promise<pb.PublishNetworkToBootstrapPeerResponse> {
    return this.host.expectOne(
      this.host.call(
        "Bootstrap/PublishNetworkToPeer",
        new pb.PublishNetworkToBootstrapPeerRequest(arg)
      )
    );
  }
}
